### PR TITLE
feat: allow changing default command timeout

### LINKED ISSUE #2243: Option to disable or set timeout
#### Is your feature request related to a problem? Please describe.
Ability to configure or disable timeout.

(From what I can tell) On an M1 MBP running a terminal under Rosetta, starting up node and ruby are slow enough to always trigger the timeout:

```
[WARN] - (starship::utils): Executing command "node" timed out
[WARN] - (starship::utils): Executing command "ruby" timed out
```

`explain` seems to show them working:

```
 Here's a breakdown of your prompt:
 codex  (4ms)                -  The current working directory
 on main  (<1ms)             -  The active branch of the repo in your current directory
 via node v14.15.3  (378ms)  -  The currently installed version of NodeJS
 via ruby v2.7.2  (344ms)    -  The currently installed version of Ruby
 at 02:29:36 AM  (<1ms)      -  The current local time
 ❯  (<1ms)                   -  A character (usually an arrow) beside where the text is entered in your terminal
```

#### Describe the solution you'd like
Unless this is some kind of M1-related bug, it seems like being able to disable or increase the timeout would fix the problem.

#### Describe alternatives you've considered
Turning off the extra information, but that mostly defeats the purpose.

### PR DESCRIPTION
**Pull request by davidkna**
#### Description

This PR adds a new global configuration option `command_timeout` which allows changing the default command timeout. This is mainly achieved by moving most of the usage of `utils::exec_cmd` to `Context`. This should also allow for dynamic command mocking in the future.

#### Motivation and Context

Closes #2243
Related #942

#### How Has This Been Tested?

- [x] I have tested using **MacOS**
- [ ] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:

- [x] I have updated the documentation accordingly.
- [ ] I have updated the tests accordingly.

### CODE REVIEW

**Review by andytom — APPROVED** [review]
LGTM

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -51,20 +50,19 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
...
 ```rust
 use super::{Context, Module, RootModuleConfig};
 
 use crate::configs::php::PhpConfig;
 use crate::formatter::StringFormatter;
-use crate::utils;
 
 
 pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
    // Here `my_env_var` will be either the stdout of the called command or the function
```
Comment (andytom) [inline CONTRIBUTING.md:63]: Not a blocker but worth updating this while we are editing this file
```suggestion
   // Here `output` will be either the stdout of the called command or the function
```

**Review by Snuggle — APPROVED** [review]
LGTM!

**Comment by jenhausu** [comment]
Is this feature updated to the document? I can't find it.

**Comment by andytom** [comment]
> Is this feature updated to the document? I can't find it.

Hi @jenhausu, no this hasn't been documented but we have an open issue, #2544, to add it to the docs.

**Pull request merged**